multiverse/opinion_manager.go

```diff
@@ -105,6 +106,11 @@ func (o *OpinionManager) UpdateWeights(messageID MessageID) (updated bool) {
 		// We calculate the approval weight of the branch based on the node who issued the message to the branch (i.e., it already voted for the branch).
 		o.approvalWeights[lastOpinion.Color] -= o.tangle.WeightDistribution.Weight(message.Issuer)
 		o.events.ApprovalWeightUpdated.Trigger(lastOpinion.Color, int64(-o.tangle.WeightDistribution.Weight(message.Issuer)))
+
+		// Record the min confirmed weight
+		if o.checkColorConfirmed(lastOpinion.Color) {
```

Comment:
`o.checkColorConfirmed` will return `true` only when the Color's accumulated weight is above the confirmation threshold. We should either store the node's currently confirmed color in the local OpinionManager, or compare `lastOpinion.Color` with `maxOpinion := getMaxOpinion(o.approvalWeights)`. However I'm not sure if the latter would behave correctly for an edge case when the color is unconfirmed.

Reply:
what about something like this?
`if o.colorConfirmed && o.ownOpinion == lastOpinion.Color && !o.chekColorConfirmed(o.ownOpinion)`

Reply:
Good idea! After several runs and manual investigation it can work correctly.